packages/mds-stream-processor/connectors/kafka-connector.ts

```diff
@@ -80,7 +81,130 @@ export const KafkaSource =
+          return reduced
+        }, [])
+      : [messages]
+
+  if (chunks.length > 1) {
+    StreamProcessorLogger.info(`Splitting ${messages.length} messages into ${chunks.length} chunks for processing`)
+  }
+  return chunks
+}
+/*
+ * KafkaBatchSource is a KafkaSource that batches messages into chunks of size `batchSize`
+ * and sends them to the processor.
+ * It is useful for processing large amounts of data in batches,
+ * but it is not recommended for processing small amounts of data.
+ * It is also not recommended for processing data that is not in JSON format.
+ *
+ * @param topics - The topics to consume from
+ * @param batchSize - The number of messages to batch together and send to the processor
+ * @param options - Options for the KafkaStreamConsumer
+ * @returns A StreamSource that batches messages into chunks of size `batchSize` and sends them to the processor
+ */
+export const KafkaBatchSource =
+  <TMessage>(
+    topics: SingleOrArray<string>,
+    {
+      messageLogger,
+      healthStatus,
+      ...options
+    }: Partial<KafkaStreamConsumerOptions & { messageLogger: (messages: TMessage[]) => string | undefined }> & {
+      healthStatus: HealthStatus
+    },
+    batchSize?: number
+  ): StreamSource<TMessage[]> =>
+  processor => {
+    StreamProcessorLogger.info('Creating KafkaSource', { topics, options })
+    return stream.KafkaStreamConsumer(
+      topics,
       healthStatus,
+      undefined,
+      async batchPayload => {
+        const {
+          batch: { messages, topic, partition, lastOffset },
+          resolveOffset,
+          heartbeat,
+          commitOffsetsIfNecessary,
+          isRunning,
+          isStale
+        } = batchPayload
+
+        /* 1. Chunk the messages into chunks of size `batchSize`` */
+        /* 2. Check if the batch is not stale and if processor is still running */
+        /* 3. Parse All Message values */
+        /* 4. Process All Messages */
+        /* 5. Resolve Offsets */
+        /* 6. Commit Offsets */
+        /* 7. Send Heartbeat */
+
+        for (const messagesChunk of asChunks(messages, batchSize)) {
+          if (!isStale() && isRunning()) {
+            const { messages: parsedMessages, errors: parseErrors } = messagesChunk.reduce(
+              (acc, { value }) => {
+                const messageOrError: TMessage | ParseError = (() => {
+                  try {
+                    if (value) return JSON.parse(value.toString())
+                  } catch (err) {
+                    return new ParseError('JSON.parse() failure', value)
+                  }
+                })()
+                if (messageOrError instanceof ParseError) {
+                  acc.errors.push(messageOrError)
+                } else {
+                  acc.messages.push(messageOrError)
+                }
+                return acc
+              },
+              { messages: [] as TMessage[], errors: [] as ParseError[] }
+            )
```

Comment:
I'd use generics for this instead of `as` casts in the reducer initializer
```suggestion
            const { messages: parsedMessages, errors: parseErrors } = messagesChunk.reduce<{ messages: TMessage[], errors: ParseError[]}>(
              (acc, { value }) => {
                const messageOrError: TMessage | ParseError = (() => {
                  try {
                    if (value) return JSON.parse(value.toString())
                  } catch (err) {
                    return new ParseError('JSON.parse() failure', value)
                  }
                })()
                if (messageOrError instanceof ParseError) {
                  acc.errors.push(messageOrError)
                } else {
                  acc.messages.push(messageOrError)
                }
                return acc
              },
              { messages: [] , errors: [] }
            )
```